docs/visualizing-your-sql-queries/plotting-with-ggplot.md

```diff
@@ -1,14 +1,638 @@
+
+```{code-cell} ipython3
+ax = %sqlplot histogram --table sqlplot_boxplot_example --column payments average_salary --bins 25
+ax.set_title("Histogram of Loan Payments and Average Salary ($)")
+ax.set_xlabel("Amount ($)")
+```
+
+### Question 1 (Easy)
+
++++
+
+The finance manager is impressed with how quickly you produced these plots! He wants another easy deliverable: facet the multiple column boxplot and histogram in a single graph. Make sure to customize the plot with clear axes labels and titles!
+
+<b>Hint:</b> Recall the seaborn [tutorial](https://ploomber-sql.readthedocs.io/en/latest/visualizing-your-sql-queries/plotting-with-seaborn.html#id2) of how to add/position multiple matplotlib axes subplots.
+
+<!-- #region -->
+<details>
+<summary>Answers</summary>
+
+We can use the same CTE from the example above as well as `%sqlplot` functions for the individual visualizations. The only change is that we employ `plt.subplots()` for each `%sqlplot` call and use base matplotlib functions for customizing labels:
+
+```{code-cell} ipython3
+plt.figure(figsize=(12, 3), dpi=300)  # Initialize blank canvas
+
+plt.subplot(1, 2, 1)  # first quadrant
+%sqlplot boxplot --table sqlplot_boxplot_example --column payments average_salary --orient h
+plt.title("Boxplot of Loan Payments and Average Salary ($)")  # Set title
+plt.xlabel("Loan Amount ($)")  # Set x-axis label
+
+plt.subplot(1, 2, 2)  # second quadrant
+%sqlplot histogram --table sqlplot_boxplot_example --column payments average_salary --bins 25
+plt.title("Histogram of Loan Payments and Average Salary ($)")  # Set title
+plt.xlabel("Loan Amount ($)")  # Set x-axis label
+```
+
+</details>
+<!-- #endregion -->
+
+<!-- #region -->
+
+```{important}
+Initializing the whole figure with `fig` and assigning individual axes with `ax1` and `ax2`, [like in this example](https://ploomber-sql.readthedocs.io/en/latest/visualizing-your-sql-queries/plotting-with-seaborn.html#question-3-easy), would not work.
+```
+
+## `ggplot` API
+
+You might be wondering "how can ggplot, the R package, function in Jupyter Notebooks?". Do not worry! In this tutorial, we will be learning about JupySQL's `ggplot` API to visualize our SQL queries. This plotting technique will be useful for avid R programmers, who are familiar with `ggplot2`, and for first-time learners.
+
+The `ggplot` API is built on top of `matplotlib` and is structured around the principles of the Grammar of Graphics, allowing you to build any graph using the same `ggplot2` components: a data set, a coordinate system, and geoms (geometric objects). However, to make the API suitable for JupySQL, we <b>input a SQL table name, instead of a dataset</b>. Therefore, the same [workflow](https://ploomber-sql.readthedocs.io/en/latest/visualizing-your-sql-queries/plotting-with-seaborn.html#barplots) of creating our CTE, which was employed in the `seaborn` tutorial, will be in action here as well (no need to convert the CTE into a pandas `DataFrame()`).
```

Comment:
Should we have some reference to the "Grammar of Graphics"? Maybe a hyperlink?

Reply:
I found this bookdown link:
https://bookdown.org/alhdzsz/data_viz_ir/ggbasics.html